pkg/ddevapp/config.go

```diff
@@ -1385,28 +1387,37 @@ func WriteImageDockerfile(fullpath string, contents []byte) error {
 func HasAllowedLocation(app *DdevApp) error {
 	homeDir, _ := os.UserHomeDir()
 	if app.AppRoot == homeDir || app.AppRoot == filepath.Dir(globalconfig.GetGlobalDdevDir()) {
-		return fmt.Errorf("'ddev config' is not allowed in your home directory (%v)", app.AppRoot)
+		return fmt.Errorf("a project is not allowed in your home directory (%v)", app.AppRoot)
 	}
 	rel, err := filepath.Rel(app.AppRoot, homeDir)
 	if err == nil && !strings.HasPrefix(rel, "..") {
-		return fmt.Errorf("'ddev config' is not allowed in the parent directory of your home directory (%v)", app.AppRoot)
+		return fmt.Errorf("a project is not allowed in the parent directory of your home directory (%v)", app.AppRoot)
 	}
 	rel, err = filepath.Rel(globalconfig.GetGlobalDdevDir(), app.AppRoot)
 	if err == nil && !strings.HasPrefix(rel, "..") {
-		return fmt.Errorf("'ddev config' is not allowed in your global config directory (%v)", app.AppRoot)
+		return fmt.Errorf("a project is not allowed in your global config directory (%v)", app.AppRoot)
 	}
 	if fileutil.FileExists(filepath.Join(app.AppRoot, "cmd/ddev/main.go")) && fileutil.FileExists(filepath.Join(app.AppRoot, "cmd/ddev_gen_autocomplete/ddev_gen_autocomplete.go")) {
-		return fmt.Errorf("'ddev config' is not allowed in the directory used for DDEV development (%v)", app.AppRoot)
-	}
-	projectList := globalconfig.GetGlobalProjectList()
+		return fmt.Errorf("a project is not allowed in the directory used for DDEV development (%v)", app.AppRoot)
+	}
+	projectMap := globalconfig.GetGlobalProjectList()
+	projectList := make([]*globalconfig.ProjectInfo, 0, len(projectMap))
+	for _, project := range projectMap {
+		projectList = append(projectList, project)
+	}
+	// Sort the projects by AppRoot in reverse alphabetical order,
+	// so deeper-level projects (longer paths) take precedence
+	sort.Slice(projectList, func(i, j int) bool {
+		return projectList[i].AppRoot > projectList[j].AppRoot
+	})
```

Comment:
Sorting is needed because if a project in the parent directory is returned before another project in the subdirectory, it will not show any warning.